pilot/api/data.py

```diff
@@ -506,6 +631,11 @@ def transfer_files(self, copytool, files, activity, **kwargs):
         # check if files exist before actual processing
         # populate filesize if need, calc checksum
         for fspec in files:
+            self.logger.info("To transfer file(activity: %s): %s" % (activity, fspec))
+            if not fspec.is_user_defined_ddmendpoint:
```

Comment:
OK.. you see since you have mixed the workflow, you need to apply special if-case-hacks..
Why not to extend the function for proper OS-based client and do all initializations properly, without storing redundant `storage_id`, `is_user_defined_ddmendpoint` and etc.. 
as result we get clear definition, clear workflow and real FileSpec model without internal fields used only to apply hacks